tutorials/introductory/patchwork_axes.py

```diff
@@ -0,0 +1,195 @@
+"""
+==============
+Patchwork Axes
+==============
+
+This is inspired by the `patchwork
+<https://github.com/thomasp85/patchwork>`__ library for R.  While we
+do not implement the operator overloading style, we do provide a
+pythonic API for specifying (nested) Axes layouts.  If you need to do
+something that is not clearly expressed using this method, please see
+:ref:`sphx_glr_tutorials_intermediate_gridspec.py`.
+
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def label_axes(ax_dict):
+    for k, ax in ax_dict.items():
+        ax.set_title(k, loc="left")
+    return ax_dict
+
+
+###############################################################################
+# Express String specification
+# ============================
+#
+# Using single leters we can "draw" the axes we want.  The following
+
+
+layout = """
+AAC
+BBC
+"""
+
+###############################################################################
+# will give us 3 axes.  Two will be stacked vertically in the left and
+# one will span the vertical space on the right.  The Axes on the left
+# will be twice as wide as the Axes on the right and the Axes on the
+# right will be twice as tall as Axes on the left.  But, as they say
+# a picture is worth a thousand words:
+
+fig = plt.figure(constrained_layout=True)
+axd = fig.build_grid(layout)
+label_axes(axd)
+
+###############################################################################
+# The returned value from `.Figure.build_grid` is a dictionary mapping the
+# label you set to the axes
+
+print(axd)
+
+###############################################################################
+# If instead we wanted a 2x2 grid on the left and the same span on the right
+axd = plt.figure(constrained_layout=True).build_grid(
+    """
+ABC
+DEC
+"""
```

Comment:
do we want to pass the string through `inspect.cleandoc` to avoid such awkward indents?